docs/latest/design/envoy-patch-policy.md

```diff
@@ -0,0 +1,150 @@
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io/v1beta1
+    kind: Gateway
+    name: eg-gw
+    namespace: default
+  type: JSONPatch
+  jsonPatches:
+    - type: "type.googleapis.com/envoy.config.listener.v3.Listener"
+      # The listener name is of the form <GatewayNamespace>-<GatewayName>-<GatewayListenerName>
+      name: default-eg-gw-example
+      operation:
+        op: add
+        path: "/filter_chains/0/filters/0/http_filters/0"
+        value: |
+          name: "envoy.filters.http.ratelimit"
+          typed_config:
+            "@type": "type.googleapis.com/envoy.extensions.filters.http.ratelimit.v3.RateLimit"
+            domain: "eag-ratelimit"
+            failure_mode_deny: true
+            timeout: 1s
+            rate_limit_service:
+              grpc_service:
+                envoy_grpc:
+                  cluster_name: rate-limit-cluster
+              transport_api_version: V3
+    - type: "type.googleapis.com/envoy.config.route.v3.RouteConfiguration"
+      # The route name is of the form <GatewayNamespace>-<GatewayName>-<GatewayListenerName>
+      name: default-eg-gw-example
+      operation:
+        op: add
+        path: "/virtual_hosts/0/rate_limits"
+        value: |
+          - actions:
+              - remote_address: {}
+    - type: "type.googleapis.com/envoy.config.cluster.v3.Cluster" 
+      name: rate-limit-cluster
+      operation:
+        op: add
+        path: "/"
+        value: |
+          name: rate-limit-cluster
+          type: STRICT_DNS
+          connect_timeout: 10s
+          lb_policy: ROUND_ROBIN
+          http2_protocol_options: {}
+          load_assignment:
+            cluster_name: rate-limit-cluster
+            endpoints:
+              - lb_endpoints:
+                  - endpoint:
+                      address:
+                        socket_address:
+                          address: ratelimit.svc.cluster.local
+                          port_value: 8081        
+```
+
+
+## Verification
+* Offline - Leverage [egctl x translate][] to ensure that the `EnvoyPatchPolicy` can be successfully applied and the desired
+output xDS is created.
+* Runtime - Use the `Status` field within `EnvoyPatchPolicy` to highlight whether the patch was applied successfully or not.
+
+## State of the World
+* Istio - Supports the [EnvoyFilter][] API which allows users to customize the output xDS using patches and proto based merge
+semantics.
+
+## Design Decisions
+* This API will only support a single `targetRef` and can bind to only a `Gateway` resource. This simplifies reasoning of how
+patches will work.
+* This API will always be an experimental API and cannot be graduated into a stable API because Envoy Gateway cannot garuntee
+  * that the naming scheme for the generated resources names will not change across releases
+  * that the underlying Envoy Proxy API will not change across releases
+
+## Open Questions
+* Should this be an Opt In API disabled by default ?
```

Comment:
100%. One should *definitely* have to explicitly opt in to using fragile, experimental APIs.

Reply:
raised https://github.com/envoyproxy/gateway/issues/1468 to track this